src/coreclr/jit/emitloongarch64.cpp

```diff
@@ -0,0 +1,6431 @@
+
+            code |= ((imm >> 18) & 0x1f);        // offs[20:16]
+            code |= reg << 5;                    // rj
+            code |= ((imm >> 2) & 0xffff) << 10; // offs[15:0]
+            break;
+        case INS_movfr2cf:
+            assert(isFloatReg(reg));
+            assert((0 <= imm) && (imm <= 7));
+
+            code |= (reg & 0x1f) << 5; // fj
+            code |= imm /*& 0x7*/;     // cc
+            break;
+        case INS_movcf2fr:
+            assert(isFloatReg(reg));
+            assert((0 <= imm) && (imm <= 7));
+
+            code |= (reg & 0x1f);         // fd
+            code |= (imm /*& 0x7*/) << 5; // cc
+            break;
+        case INS_movgr2cf:
+            assert(isGeneralRegister(reg));
+            assert((0 <= imm) && (imm <= 7));
+
+            code |= reg << 5;      // rj
+            code |= imm /*& 0x7*/; // cc
+            break;
+        case INS_movcf2gr:
+            assert(isGeneralRegister(reg));
+            assert((0 <= imm) && (imm <= 7));
+
+            code |= reg;                  // rd
+            code |= (imm /*& 0x7*/) << 5; // cc
+            break;
+        default:
+            unreached();
+            break;
+    } // end switch (ins)
+
+    instrDesc* id = emitNewInstr(attr);
+
+    id->idIns(ins);
+    id->idReg1(reg);
+    id->idAddr()->iiaSetInstrEncode(code);
+
+    id->idCodeSize(4);
+    // dispIns(id);
+    appendToCurIG(id);
+}
+
+//------------------------------------------------------------------------
+// emitIns_Mov: Emits a move instruction
+//
+// Arguments:
+//    ins       -- The instruction being emitted
+//    attr      -- The emit attribute
+//    dstReg    -- The destination register
+//    srcReg    -- The source register
+//    canSkip   -- true if the move can be elided when dstReg == srcReg, otherwise false
+//    insOpts   -- The instruction options
+//
+void emitter::emitIns_Mov(
+    instruction ins, emitAttr attr, regNumber dstReg, regNumber srcReg, bool canSkip, insOpts opt /* = INS_OPTS_NONE */)
+{ // TODO-LoongArch64: should amend for LoongArch64/LOONGARCH64.
+    assert(IsMovInstruction(ins));
+
+    if (!canSkip || (dstReg != srcReg))
+    {
+        if ((EA_4BYTE == attr) && (INS_mov == ins))
+            emitIns_R_R_I(INS_slli_w, attr, dstReg, srcReg, 0);
+        else
+            emitIns_R_R(ins, attr, dstReg, srcReg);
+    }
+}
+
+/*****************************************************************************
+ *
+ *  Add an instruction referencing two registers
+ */
+
+void emitter::emitIns_R_R(
+    instruction ins, emitAttr attr, regNumber reg1, regNumber reg2, insOpts opt /* = INS_OPTS_NONE */)
+{
+    code_t code = emitInsCode(ins);
+
+    if (INS_mov == ins)
+    {
+        assert(isGeneralRegisterOrR0(reg1));
+        assert(isGeneralRegisterOrR0(reg2));
+        code |= reg1;      // rd
+        code |= reg2 << 5; // rj
+    }
+    else if ((INS_ext_w_b <= ins) && (ins <= INS_cpucfg))
+    {
+        // case INS_ext_w_b:
```

Comment:
Should these comments be asserts? (i.e., `assert(ins == INS_ext_w_b || ins == ...)`)

Reply:
OK, Thanks.